src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkImageService.kt

```diff
@@ -1,19 +1,45 @@
 package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.prisonperson.identifyingmarks
 
+import org.springframework.dao.DuplicateKeyException
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.DuplicateMappingException
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.prisonperson.identifyingmarks.api.IdentifyingMarkImageMappingDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.prisonperson.identifyingmarks.api.toDto
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.prisonperson.identifyingmarks.api.toEntity
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.service.NotFoundException
 import java.util.UUID
 
 @Service
 class IdentifyingMarkImageService(
   private val repository: IdentifyingMarkImageMappingRepository,
 ) {
-  suspend fun getIdentifyingMarkImageMapping(nomisOffenderImageId: Long): IdentifyingMarkImageMapping? =
+  suspend fun getIdentifyingMarkImageMapping(nomisOffenderImageId: Long): IdentifyingMarkImageMappingDto =
     repository.findById(nomisOffenderImageId)
+      ?.toDto()
+      ?: throw NotFoundException("Identifying mark image mapping not found for NOMIS offender image id $nomisOffenderImageId")
 
-  suspend fun getIdentifyingMarkImageMapping(dpsImageId: UUID): IdentifyingMarkImageMapping? =
+  suspend fun getIdentifyingMarkImageMapping(dpsImageId: UUID): IdentifyingMarkImageMappingDto =
     repository.findByDpsId(dpsImageId)
+      ?.toDto()
+      ?: throw NotFoundException("Identifying mark image mapping not found for DPS image id $dpsImageId")
 
-  suspend fun createIdentifyingMarkImageMapping(mapping: IdentifyingMarkImageMapping) {
-    repository.save(mapping)
+  suspend fun createIdentifyingMarkImageMapping(mapping: IdentifyingMarkImageMappingDto) {
+    try {
+      repository.save(mapping.toEntity())
+    } catch (e: DuplicateKeyException) {
```

Comment:
Interesting, I thought that didn't work - something to do with commits and reactive - very odd